pkg/awsutils/awsutils.go

```diff
@@ -224,8 +225,66 @@ func prometheusRegister() {
 	}
 }
 
+//StringSet is a set of strings
+type StringSet struct {
+	sync.RWMutex
+	data 	sets.String
+}
+
+func (ss *StringSet) AWSStrings() []*string {
+	ss.RLock()
+	defer ss.RUnlock()
+	var dataSlice []*string
+	for key, _ := range ss.data {
+		dataSlice = append(dataSlice, aws.String(key))
+	}
+	return dataSlice
+}
+
+func (ss *StringSet) Set(items []string) {
+	ss.Lock()
+	defer ss.Unlock()
+	for _, item := range items {
+		ss.data.Insert(item)
+	}
+}
+
+func (ss *StringSet) IsEmpty() bool {
+	if ss.data.Len() == 0 {
+		ss.data = sets.NewString()
+		return true
+	}
+	return false
+}
+
+func (ss *StringSet) compareAddDel (items []string) {
+	newSet := StringSet{data: sets.String{}}
+	newSet.Set(items)
+	//For example: s1 = {a1, a2, a3} s2 = {a1, a2, a4, a5} s1.Difference(s2) = {a3} s2.Difference(s1) = {a4, a5}
+	setDiff := ss.data.Difference(newSet.data)
+	if setDiff.Len() != 0 {
+		ss.Lock()
+		defer ss.Unlock()
```

Comment:
I would expect the `defer ss.Unlock()` here to not run until the whole `compareAddDel()` function exits, so if we both add and delete an item, this would deadlock. Probably better to just do it once in the beginning of this function.

Maybe worth having a test that does this? Create a set of 3 strings, `["a1", "a2", "a3"}`, then call it with `["a2", "a3", "a4"]`.